### DISCUSSION TITLE
How to get discovery for buttons? MQTT showing /RESULT {"Button2":{"Action":"SINGLE"}} but no sensor discovery.

### DISCUSSION THREAD

**Discussion by markfickett** [arendst/Tasmota#19887] [Support request] [answered]
I wired up a few momentary buttons with an external pullup to my Tasmota device. I set `SetOption73 1` and `ButtonTopic 1`, and I see MQTT messages like `stat/tasmota_884477/RESULT = {"Button2":{"Action":"SINGLE"}}`. However, I don’t see the buttons mentioned in the discovery topic (nor do they show up as entities in HomeAssistant). I do see my other sensors and switch (from a PIR): `tasmota/discovery/E89F6D884477/sensors {"sn":{"Time":"2023-10-31T15:49:07","Switch1":"OFF","AM2301":{"Temperature":22.2,"Humidity":38.5,"DewPoint":7.4},"TempUnit":"C"},"ver":1}`. The buttons are configured for example as `D6 = Button_n 3`.

What configuration do I need in Tasmota so that my buttons get autodiscovered? Is that possible?

I did see this related discussion: [Added Button entity not discovered](https://github.com/arendst/Tasmota/discussions/17215) but I can't tell if it's describing this situation or something more complex.

**Comment by barbudor** [comment, marked as answer]
There is no discovery for Buttons (only for the Relay they are supposed to be attached to) 
If I'm not wrong, there are for Switches

In most of the cases Switches are more flexible than Buttons... except for the lack of multipress
